client/core/core.go

```diff
@@ -2500,6 +2824,23 @@ func (c *Core) Login(pw []byte) (*LoginResult, error) {
 		return nil, fmt.Errorf("cannot log in because app has not been initialized")
 	}
 
+	// Check for necessary authenticated updates.
+	coreVersion, err := c.db.CoreVersion()
+	if err != nil {
+		return nil, fmt.Errorf("CoreVersion error: %v", err)
+	}
+
+	switch coreVersion {
+	case 0:
+		err := c.upgradeCoreToV1(pw)
```

Comment:
Similar to the unauthenticated upgrades, I'd like if we could make a backup db file first.